### PR TITLE
feat: use golang.org/x/mod to parse `go.mod` file

### PR DESCRIPTION
**Pull request by j178**
Use `golang.org/x/mod` to do a formal parsing of `go.mod` file.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -60,12 +62,21 @@ func main() {
 }
 
 func pkgsInfo(r io.Reader, cache map[string]string) error {
+	data, err := io.ReadAll(r)
```
Comment (tebeka) [inline main.go]: Can  you add an `io.LimitReader` here (say 1MB)? In the scanner version we go line by line so I don't care much about size.

**Review by tebeka — COMMENTED** [review]
Thanks! Just a tiny comment. If you don't have time to implement it - I'll merge and do it myself.

File: main.go (hunk 1 above)
Comment (j178) [reply main.go]: Makes sense. Per the module proxy doc, go.mod files are limited to 16 MiB actually.
https://go.dev/ref/mod#zip-path-size-constraints

File: main.go [hunk 2]
```diff
@@ -60,12 +62,21 @@ func main() {
 }
 
 func pkgsInfo(r io.Reader, cache map[string]string) error {
+	data, err := io.ReadAll(io.LimitReader(r, 1<<24)) // go.mod files are limited to 16 MiB
```
Comment (tebeka) [inline main.go:65]: I'm approving. For readability I'll rewrite it to `16*(1<<20)` :)

**Pull request merged**